### PR TITLE
Delay warnings until after plugins

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: Rivet/Functions/Convert-FileInfoToMigration.ps1 [hunk 1]
```diff
@@ -34,63 +34,63 @@ function Convert-FileInfoToMigration
...
-                        [Parameter(Mandatory=$true,ValueFromPipeline=$true)]
-                        [object]
-                        # The migration object to invoke.
-                        $Operation,
-
-                        [Parameter(ParameterSetName='Push',Mandatory=$true)]
-                        [Collections.Generic.List[Rivet.Operations.Operation]]
-                        [AllowEmptyCollection()]
-                        $OperationsList,
+        filter Add-Operation
```
Comment (pnstickne) [inline Rivet/Functions/Convert-FileInfoToMigration.ps1:37]: `filter`? That's a new one to me >_>

File: Rivet/Functions/Convert-FileInfoToMigration.ps1 [hunk 2]
```diff
@@ -34,63 +34,63 @@ function Convert-FileInfoToMigration
...
 
-                            $_
+                $OperationsList.AddRange($operations)
+            }
+        }
 
-                            Invoke-RivetPlugin -Event ([Rivet.Events]::AfterOperationLoad) -Parameter $pluginParameter
+        foreach( $fileInfo in $InputObject )
+        {
+            $dbName = Split-Path -Parent -Path $fileInfo.FullName
```
Comment (pnstickne) [inline Rivet/Functions/Convert-FileInfoToMigration.ps1:77]: Comment above for intent would make this clearer to me. The multiple re-assignment from similar operations..

**Review by pnstickne — APPROVED** [review]

File: Rivet/Functions/Convert-FileInfoToMigration.ps1 (hunk 1 above)
Comment (splatteredbits) [reply Rivet/Functions/Convert-FileInfoToMigration.ps1:37]: It's shorthand for functions that take input in a pipeline:

```
function Name
{
    begin
    {
        # Called before first item is piped to function.
    }
    process
    {
        # Called for each item piped to the function.
    }
    end
    {
        # Called after last item is piped to function
    }
}
```

Or, with a filter:
```
filter Name
{
    # Called for each item piped to the function.
}
```

File: Rivet/Functions/Repair-Operation.ps1 [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                $warningMsg = "Unique key constraint names will be required in a future version of Rivet. Please add " +
+                              "a ""Name"" parameter (with a value of ""$($name)"") to the Add-UniqueKey " +
+                              "operation on the $($tableDesc) table's $($columnDesc) column$($pluralSuffix)."
+            }
+            'RemoveDefaultConstraint'
+            {
+                $name = New-ConstraintName -Default -SchemaName $schemaName -TableName $tableName -ColumnName $columnName
+                $warningMsg = "Default constraint names will be required in a future version of Rivet. Please add a " +
+                              """Name"" parameter (with a value of ""$($name)"") to the Remove-DefaultConstraint " +
+                              "operation for the $($tableDesc) table's ""$($columnDesc)"" column."
```
Comment (pnstickne) [inline Rivet/Functions/Repair-Operation.ps1:85]: Different formatting from others.

File: Rivet/Functions/Repair-Operation.ps1 [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                              "add a ""Name"" parameter (with a value of ""$($name)"") to the Remove-ForeignKey " +
+                              "operation for the $($tableDesc) table that references the " +
+                              "[$($Operation.ReferencesSchemaName)].[$($Operation.ReferencesTableName)] table."
+            }
+            'RemoveIndexOperation'
+            {
+                $name = New-ConstraintName -Index -SchemaName $schemaName -TableName $tableName -ColumnName $columnName -Unique:$Operation.Unique
+                $warningMsg = "Index names will be required in a future version of Rivet. Please add a ""Name"" " +
+                              "parameter (with a value of ""$($name)"") to the Remove-Index operation for the " +
+                              "$($tableDesc) table's ""$($columnDesc)"" column$($pluralSuffix)."
```
Comment (pnstickne) [inline Rivet/Functions/Repair-Operation.ps1:104]: Different formatting (and a few others).

File: Rivet/Functions/Repair-Operation.ps1 [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        {
+            Write-Warning -Message $warningMsg
+        }
+
+        return $Operation
+    }
+
+    end
+    {
+
```
Comment (pnstickne) [inline Rivet/Functions/Repair-Operation.ps1:134]: Extra newline.

File: Source/Operations/RemoveIndexOperation.cs [hunk 6]
```diff
@@ -4,11 +4,17 @@ namespace Rivet.Operations
...
 	{
-		public RemoveIndexOperation(string schemaName, string tableName, string name)
+		public RemoveIndexOperation(string schemaName, string tableName, string name, string[] columnName, bool unique)
 			: base(schemaName, tableName, name)
 		{
+			ColumnName = columnName;
+			Unique = unique;
 		}
 
+		public string[] ColumnName { get;  }
```
Comment (pnstickne) [inline Source/Operations/RemoveIndexOperation.cs:14]: Prefer IReadOnlyList/IReadOnlyCollection when practical.

File: Rivet/Functions/Repair-Operation.ps1 (hunk 3 above)
Comment (splatteredbits) [reply Rivet/Functions/Repair-Operation.ps1:85]: I don't see what's different.

File: Rivet/Functions/Repair-Operation.ps1 (hunk 4 above)
Comment (splatteredbits) [reply Rivet/Functions/Repair-Operation.ps1:104]: Again, I don't see what's different.

File: Source/Operations/RemoveIndexOperation.cs (hunk 6 above)
Comment (splatteredbits) [reply Source/Operations/RemoveIndexOperation.cs:14]: These are part of Rivet's API and each property needs to be modifiable for user's in their plug-ins.

**Pull request merged**